stratum/hal/lib/common/yang_parse_tree_paths.cc

```diff
@@ -2818,6 +2818,20 @@ void SetUpComponentsComponentName(const std::string& name, TreeNode* node) {
       ->SetOnTimerHandler(poll_functor);
 }
 
+////////////////////////////////////////////////////////////////////////////////
+// /components/component[name=<name>]/state/type
+void SetUpComponentsComponentStateType(
+    const std::string& name, TreeNode* node) {
+  auto poll_functor = [name](const GnmiEvent& /*event*/,
+                             const ::gnmi::Path& path,
+                             GnmiSubscribeStream* stream) {
+    return SendResponse(GetResponse(path, "OPTICAL_CHANNEL"), stream);
+  };
+
+  node->SetOnPollHandler(poll_functor)
+      ->SetOnTimerHandler(poll_functor);
+}
```

Comment:
Add new parameter `std::string& type` and pass this parameter to `SendResponse` call. So we can reuse this function in the future for other components.

Reply:
Fixed. Please review.